src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2012,8 +2021,38 @@ private bool IsTypeDeclarationStart()
             }
         }
 
+        public MemberDeclarationSyntax ParseMemberDeclaration()
+        {
+            // Creates a dummy declaration node to which we can attach a stack overflow message
+            MemberDeclarationSyntax createEmptyNodeFunc()
+            {
+                return _syntaxFactory.NamespaceDeclaration(
+                    SyntaxFactory.MissingToken(SyntaxKind.NamespaceKeyword),
+                    CreateMissingIdentifierName(),
+                    SyntaxFactory.MissingToken(SyntaxKind.OpenBraceToken),
+                    new SyntaxList<ExternAliasDirectiveSyntax>(),
+                    new SyntaxList<UsingDirectiveSyntax>(),
+                    new SyntaxList<MemberDeclarationSyntax>(),
+                    SyntaxFactory.MissingToken(SyntaxKind.CloseBraceToken),
+                    semicolonToken: null);
+            }
+            return ParseWithStackGuard(
+                () => this.ParseMemberDeclarationOrStatement(SyntaxKind.StructDeclaration),
```

Comment:
>ParseMemberDeclarationOrStatement [](start = 27, length = 33)

The name indicates that it can return  something other than a member declaration

Reply:
It can, depending on its argument. It can accept global statements in interactive or script mode.

Reply:
But not when within a struct.

Reply:
> But not when within a struct.

Could you please add an explicit comment here that passing ```SyntaxKind.StructDeclaration``` also has this side-effect?

---
In reply to: [210053499](https://github.com/dotnet/roslyn/pull/29263#discussion_r210053499) [](ancestors = 210053499)